Dynamo20/Dynamo_Engine/Convert/FromDesignScript.cs

```diff
@@ -159,10 +159,18 @@ public static BHG.Line FromDesignScript(this ADG.Line line)
         // A quasi-fallback method - lines sometimes come out of Dynamo as objects of type Line, sometimes of type Curve.
         public static BHG.ICurve FromDesignScript(this ADG.Curve curve)
         {
-            if (Math.Abs(curve.StartPoint.DistanceTo(curve.EndPoint) - curve.Length) <= 1e-6)
-                return Geometry.Create.Line(curve.StartPoint.FromDesignScript(), curve.EndPoint.FromDesignScript());
+            // For some reason Dynamo wraps curves of other type into Curve, instead of using them as they are - Explode here is simply unwrapping the actual curves of more specific types.
+            if (curve.GetType() == typeof(ADG.Curve))
+            {
+                List<ADG.Curve> curves = curve.Explode().Cast<ADG.Curve>().ToList();
+                if (curves.Count != 0)
+                    return curves[0].IFromDesignScript();
+                else
+                    return new BHG.PolyCurve { Curves = curves.Select(x => x.IFromDesignScript()).ToList() };
```

Comment:
when are there no curves in the curve, and why does that lead to assuming that there are multiple?
And why are we only using the first if there are many?

Reply:
It was meant to be  `if (curves.Count == 1)`, forgot to change when refactoring the previous iteration. Fixed, should be fine now.